src/main/java/org/kitteh/irc/client/library/feature/InteractionDefaults.java

```diff
@@ -0,0 +1,164 @@
+            // Both "" and null are legal here, technically. But for KICL to perform best (and to avoid != null && string.isEmpty() checks),
+            // we enforce a requirement on null.
+            throw new IllegalArgumentException("defaultString may not be an empty string, use null instead.");
+        }
+        this.defaultString = defaultString;
+    }
+
+    /**
+     * Creates a InteractionDefaults object with predefined string to default
+     * to when an Interaction key is not set.
+     */
+    public InteractionDefaults() {
+        this.defaultString = null;
+    }
+
+    /**
+     * Sets a default message for the Interaction.
+     *
+     * Will throw an IllegalArgumentException is you set the
+     * defaultString value to "". Use null instead.
+     *
+     * Will throw a RuntimeException if you try to define a value
+     * that is already defined. This often indicates a problem and
+     * #replaceDefault should be used instead.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is provided for this Interaction
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults setDefault(Interaction key, @Nullable String defaultString) {
+        if (defaultString != null && defaultString.isEmpty()) {
+            // Both "" and null are legal here, technically. But for KICL to perform best (and to avoid != null && string.isEmpty() checks),
+            // we enforce a requirement on null.
+            throw new IllegalArgumentException("defaultString may not be an empty string, use null instead.");
+        }
+
+        if (this.defaults.get(key) != null) {
+            throw new RuntimeException("The default value for " + key.toString() + " is already set with value: " + defaults.get(key));
+        }
+
+        this.defaults.put(key, defaultString);
+
+        return this;
+    }
+
+    /**
+     * Retrieve the default string value for the specified Interaction.
+     *
+     * @param key The default Interaction key to obtain the default of
+     * @return The string of the default value, or the predefined
+     * defaultString
+     */
+    public String getDefault(Interaction key) {
+        String value = this.defaults.get(key);
+
+        return value != null ? value : this.defaultString;
+    }
+
+    /**
+     * Retrieve the default string value for the specified Interaction,
+     * but override the defaultString value with our own message if the
+     * value is not set.
+     *
+     * @param key The default Interaction key to obtain the default of
+     * @param messageIfUnset Instead of referencing to the defaultString,
+     * use this value instead if it is not set.
+     * @return The string of the default value, or the second parameter
+     * if the key is not set.
+     */
+    public String getDefault(Interaction key, String messageIfUnset) {
```

Comment:
`messageIfUnset` -> `defaultValue`
Is this parameter `null`able? If so, mark as `@Nullable`, otherwise as `@Nonnull` and `Sanity.nullCheck` it.